### DISCUSSION TITLE
Highlight coverage of all passenger stations and bus stops

### DISCUSSION THREAD

**Discussion by Finghin** [voxeltycoon/issues#835] [UI&UX]
When placing passenger stations or bus stops the areas already covered by other stations/stops should be highlighted as well, preferably in a different color or shade.

**Comment by jertje260** [comment]
I wanted this same thing, looked into modding this into the game, but since it is coded on the base BuilderTool.cs I'm not able to create this.

Should be as 'simple' as extending the `private SetTint(bool valid)` method by checking for each building in the `_tintedBuildings` if it is a `House` and `IsConnectedToVehicleStation()`

This would be the hacky way, but there might be a better way to do this.

Maybe by changing the way this information is retrieved, besides only getting the buildings to tint, also add a state to them (connected/notconnected) or something more like supplied/notsupplied (which then can also work for Warehouses to check if they have a connected station already

**Comment by krnhns** [comment]
Totally agree with this. Currently have to open each station and click logistics meaning a lot of windows could be open at once, so being able to see this with only one window would be perfect.

**Comment by DanielBarreleiro** [comment]
Continuing this topic and with what @krnhns said, only current way is by opening every station window, which causes fps to go down and blocks the screen.
![image](https://user-images.githubusercontent.com/47695358/145389166-7587230e-6e12-48f4-910a-12aa2518d047.png)

Maybe a way for this to be implemented for example: when building a new station highlight buildings that are already coloured with a different colour.
![image](https://user-images.githubusercontent.com/47695358/145391315-f8c4db98-da99-471a-a046-612e22cc1841.png)
Sorry for my great artistic skills.
